daml-lf/interpreter/src/main/scala/com/digitalasset/daml/lf/speedy/SBuiltin.scala

```diff
@@ -1251,6 +1251,28 @@ private[lf] object SBuiltin {
     }
   }
 
+  // Convert an interface `requiringIface` to another interface `requiredIface`, if
+  // the `requiringIface` requires `requiredIface`.
+  final case class SBFromRequiredInterface(
+      iface: TypeConName
+  ) extends SBuiltin(1) {
+
+    override private[speedy] def execute(
+        args: util.ArrayList[SValue],
+        machine: Machine,
+    ) = {
+      val record = getSRecord(args, 0)
+      // TODO https://github.com/digital-asset/daml/issues/11978
+      //  The lookup is probably slow. We may want to investigate way to make the feature faster.
+      machine.returnValue = machine.compiledPackages.interface.lookupTemplate(record.id) match {
```

Comment:
I think you could turn this into an `ImplementsDefRef` check, like in `SBUFetchInterface`. That should be faster, maybe?

Reply:
Ok, maybe not faster. But a little cleaner at least.

Reply:
Yea. Let's keep that under wraps for now.

Reply:
We are already doing this in `SBUFetchInterface`:

```
          machine.compiledPackages
            .getDefinition(ImplementsDefRef(actualTemplateId, ifaceId))
            .isEmpty
```

There's no downside in doing it here as well.

Reply:
In fact we have an issue for interface performance.
I change the TODO link to properly keep that under the radar.